src/main/thrift/parquet.thrift

```diff
@@ -578,7 +578,13 @@ enum BoundaryOrder {
 
 /** Data page header */
 struct DataPageHeader {
-  /** Number of values, including NULLs, in this data page. **/
+  /**
+   * Number of values, including NULLs, in this data page.
+   *
+   * If a ColumnIndex is present, a page must begin at a record
+   * boundary (repetition_level = 0). Otherwise, pages may begin
+   * within a record (repetition_level > 0).
```

Comment:
I wonder if we could instead state that records should not be split across pages, but that readers should tolerate this in the abscence of a page index or v2 data pages, as certain writers historically wrote such data?

Reply:
I agree with you that splitting pages will make the file less compatible, even though some writers historically did that. 

However, I would like to ensure there is consensus on updating the spec to say that writers *should* do something  before changing the spec